controllers/userController.js

```diff
@@ -1,28 +1,248 @@
 const helpers = require('../_helpers')
 
+const bcrypt = require('bcryptjs')
+const imgur = require('imgur')
+const IMGUR_CLIENT_ID = process.env.IMGUR_CLIENT_ID
+
+const db = require('../models')
+const User = db.User
+const Tweet = db.Tweet
+const Reply = db.Reply
+const Like = db.Like
+
 const userController = {
-  readUser: (req, res) => {
-    console.log(helpers.getUser(req).toJSON())
-    res.send(`Hello User ${helpers.getUser(req).name}!`)
+  readUser: (req, res, next) => {
+    const id = Number(req.params.id)
+    User.findOne({
+      where: { id },
+      include: [
+        Tweet,
+        { model: User, as: 'Followers' },
+        { model: User, as: 'Followings' },
+      ]
+    }).then(user => {
+      if (!user) return res.status(404).json({
+        message: `this user(id: ${id}) do not exist!`
+      })
+
+      const userObj = Object.fromEntries(Object.entries(user.toJSON()).slice(0, 11))
+      userObj.isSelf = (user.id === req.user.id)
+      userObj.isFollowed = user.Followers.map(follower => follower.id).includes(helpers.getUser(req).id)
+      userObj.tweetsCount = user.Tweets.length
+      userObj.followingsCount = user.Followings.length
+      userObj.followersCount = user.Followers.length
+      return res.json(userObj)
+    }).catch(next)
+  },
+
+  readTweets: (req, res, next) => {
+    const UserId = Number(req.params.id)
+    Tweet.findAll({
+      where: { UserId },
+      order: [['createdAt', 'DESC']],
+      include: [Reply, Like]
+    }).then(tweets => {
+      if (tweets.length < 1 || tweets === undefined)
+        return res.status(404).json({ message: `tweets of user(id: ${UserId}) do not exist!` })
```

Comment:
```suggestion
if (tweets.length < 1 || tweets === undefined)
        return res.status(404).json({ message: `tweets of user(id: ${UserId}) do not exist!` })
```

目前user沒有tweet資料和user查無此人的敘述都一樣，感覺應該要分別兩者的資訊會比較清楚?

Reply:
這邊是指此人的 tweets 不存在 `tweets of user` 

不然我改成這樣比較直觀：
```
This user(id: ${UserId}) do not post any tweets
```

感謝 feedback

Reply:
status code ，能的話不要用到已定義且常用的 code [HTTP狀態碼](https://zh.wikipedia.org/wiki/HTTP%E7%8A%B6%E6%80%81%E7%A0%81)

5XX 或是用沒有使用的 code 我覺得都可以，主要是前端能夠識別。